game/ato/loadouts.py

```diff
@@ -189,6 +202,11 @@ def default_for_task_and_aircraft(
             dcs_unit_type.load_payloads()
             payload = dcs_unit_type.loadout_by_name(name)
             if payload is not None:
+                if not cls.valid_payload(payload):
```

Comment:
Hello Raffson, 

I've got an issue with this, this exception is triggered when I click on Begin campaign :

```
2022-07-23 14:56:39,621 :: root :: ERROR :: Uncaught exception
Traceback (most recent call last):
  File "D:\Work\dcs_liberation\qt_ui\widgets\QTopPanel.py", line 150, in passTurn
    self.game.pass_turn(no_action=True)
  File "D:\Work\dcs_liberation\game\game.py", line 325, in pass_turn
    self.initialize_turn(events)
  File "D:\Work\dcs_liberation\game\game.py", line 408, in initialize_turn
    self.red.initialize_turn()
  File "D:\Work\dcs_liberation\game\coalition.py", line 186, in initialize_turn
    self.plan_missions()
  File "D:\Work\dcs_liberation\game\coalition.py", line 207, in plan_missions
    TheaterCommander(self.game, self.player).plan_missions(tracer)
  File "D:\Work\dcs_liberation\game\commander\theatercommander.py", line 83, in plan_missions
    result = self.plan(state)
  File "D:\Work\dcs_liberation\game\htn.py", line 83, in plan
    if task.preconditions_met(planning_state.state):
  File "D:\Work\dcs_liberation\game\commander\tasks\primitive\barcap.py", line 18, in preconditions_met
    return super().preconditions_met(state)
  File "D:\Work\dcs_liberation\game\commander\tasks\packageplanningtask.py", line 50, in preconditions_met
    return self.fulfill_mission(state)
  File "D:\Work\dcs_liberation\game\commander\tasks\packageplanningtask.py", line 102, in fulfill_mission
    self.package = fulfiller.plan_mission(
  File "D:\Work\dcs_liberation\game\commander\packagefulfiller.py", line 167, in plan_mission
    self.plan_flight(
  File "D:\Work\dcs_liberation\game\commander\packagefulfiller.py", line 85, in plan_flight
    if not builder.plan_flight(flight):
  File "D:\Work\dcs_liberation\game\commander\packagebuilder.py", line 57, in plan_flight
    flight = Flight(
  File "D:\Work\dcs_liberation\game\ato\flight.py", line 64, in __init__
    self.loadout = Loadout.default_for(self)
  File "D:\Work\dcs_liberation\game\ato\loadouts.py", line 193, in default_for
    return cls.default_for_task_and_aircraft(
  File "D:\Work\dcs_liberation\game\ato\loadouts.py", line 211, in default_for_task_and_aircraft
    if not cls.valid_payload(payload):
  File "D:\Work\dcs_liberation\game\ato\loadouts.py", line 142, in valid_payload
    pylons = payload["pylons"]
TypeError: list indices must be integers or slices, not str
```

To reproduce :
- start a new campaign on Battle for Golan Height map
- keep campaign default settings and defaults squadrons
- buy one of each available planes at Ramat David
 - click on begin campaign

I'm surprised mypy didn't catch this, the call to `cls.valid_payload` does seems invalid.

I think you might have missed this, since the call to `default_for_task_and_aircraft` is cached, and doesn't happen when you load an existing save.

Regards, 

Khopa

Reply:
I'll have a look at it right away since I'm waiting for pydcs to update again, hopefully a last time now 😂

Reply:
@Khopa Ok this is really weird indeed, typing suggests I should be getting a dictionary and not a list. Not sure what's going on here 🤔

Reply:
Should be resolved...

Reply:
Initially I expected these two occurrences of 'with_clsid' not to cause any problems, turns out the one in 'for_aircraft' is definitely a problem for mypy, and for the occurrence in '__setstate__' I assume it's better to guard against a None value.